lib/IDE/CodeCompletion.cpp

```diff
@@ -1823,6 +1823,27 @@ class CompletionLookup final : public swift::VisibleDeclConsumer {
     llvm_unreachable("unhandled kind");
   }
 
+  void addValueBaseName(CodeCompletionResultBuilder &Builder,
+                        DeclBaseName Name) {
+    auto NameStr = Name.userFacingName();
+    bool shouldEscapeKeywords =
+        // Special names (i.e. 'init') are always displayed as its user facing
+        // name.
+        !Name.isSpecial() &&
+        // User can write any keyword after '.' exept for non-special `init`.
+        // e.g. 'func `init`()' should be 'expe.`init`()'.
+        (!ExprType || NameStr == "init") &&
+        // 'self' and 'Self' are special
+        NameStr != "self" && NameStr != "Self";
```

Comment:
I didn't see any error writing x.Self as a property reference.

Reply:
`foo.self`, `foo.Self`, `self` and `Self` are all valid. We don't have to escape `self` and `Self` .
```swift
class C {
  func foo() {
    var `self` = 12
    let `Self` = 42
    self += Self
    print(self)
  }
}
```

Reply:
Ah, I was wrong.
```swift
class C {
   var `self` = 12
}
print(C().self) // C
print(C().`self`) // 12
```
I will update this part to clarify the intent.